future-apps/payroll/contracts/Payroll.sol

```diff
@@ -655,7 +655,41 @@ contract Payroll is EtherTokenConstant, IForwarder, IsContract, AragonApp {
             // No need to use SafeMath here as we already know _paymentAmount > accruedSalary
             currentSalaryPaid = _paymentAmount - accruedSalary;
         }
-        employee.lastPayroll = _getLastPayrollDate(_employeeId, currentSalaryPaid);
+        _updateEmployeeLastPayrollDate(_employeeId, currentSalaryPaid);
+    }
+
+    /**
+     * @dev Update the last payroll date for an employee based on the requested payment amount. If the requested amount
+     *      cannot be represented by a multiple of the employee's salary per second, it will be added as accrued salary.
+     * @param _employeeId Employee's identifier
+     * @param _paidAmount Requested amount to be paid to the employee
+     */
+    function _updateEmployeeLastPayrollDate(uint256 _employeeId, uint256 _paidAmount) internal {
+        Employee storage employee = employees[_employeeId];
+
+        uint256 timeDiff = _paidAmount.div(employee.denominationTokenSalary);
+
+        // We check if the division was perfect, and if not, take its ceiling to avoid giving away tiny amounts of
+        // salary and add the remainder to the accrued salary.
+        if (timeDiff.mul(employee.denominationTokenSalary) != _paidAmount) {
+            timeDiff = timeDiff.add(1);
+            // No need to use SafeMath here, we already now that _paidAmount is lower than its ceil
+            uint256 remainder = timeDiff.mul(employee.denominationTokenSalary) - _paidAmount;
+
+            uint256 currentAccruedSalary = employee.accruedSalary;
+            uint256 newAccruedSalary = currentAccruedSalary + remainder;
+            if (newAccruedSalary < currentAccruedSalary) {
+                newAccruedSalary = MAX_UINT256;
+            }
```

Comment:
I think we can just use `employee.accruedSalary = employee.accruedSalary.add(remainder)` here; it should be impossible for this to overflow (we always reset accrued salary first).